updatehub-sdk/src/listener.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2018, 2019, 2020 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Error, Result};
+use async_std::{
+    io::BufReader,
+    os::unix::net::{UnixListener, UnixStream},
+    prelude::*,
+};
+use log::warn;
+use std::{env, fs, path::Path, pin::Pin, process};
+
+const SDK_TRIGGER_FILENAME: &str =
+    "/usr/share/updatehub/state-change-callbacks.d/10-updatehub-sdk-statechange-trigger";
+const SOCKET_PATH: &str = "/run/updatehub-statechange.sock";
+
+type Callback = Box<dyn Fn(ListenerHandler) -> Pin<Box<dyn std::future::Future<Output = ()>>>>;
```

Comment:
Since some of the operations made on `ListenerHandle` can return a `Result` we should have a `Result` as the Future's output here as well.